src/components/tree-detail/tree-water-needs/watering-dialog.tsx

```diff
@@ -52,6 +57,12 @@ export const WateringDialog: React.FC = () => {
 				// wait for the dialog to close before expanding the last waterings
 				setTimeout(() => {
 					setIsLastWateringsExpanded(true);
+
+					// Update the map with the new watering amount
+					const treeId = useTreeStore.getState().selectedTreeId;
+					if (treeId) {
+						addTodayWatering(treeId, amount);
+					}
```

Comment:
is this located in the timeout by design, so the user can witness it change colors?

Reply:
yes, it's in the timeout for the user's satisfaction